### PR TITLE
Fix/valid recent context

### PR DESCRIPTION
**Pull request by Bluesmile82**
## Pivotal Tracker

https://www.pivotaltracker.com/story/show/175242862

## Description

Fix cases in which the latest context was not included on the currently available contexts

### CODE REVIEW

File: frontend/scripts/react-components/explore/explore.selectors.js [hunk 1]
```diff
@@ -122,13 +122,17 @@ const getRecentCard = () => {
...
 }
 
-const getCardsWithRecentCard = createSelector(
-  [getSankeyCards, getRecentCard],
-  (cards, recentCard) => {
+export const getCardsWithRecentCard = createSelector(
+  [getSankeyCards, getRecentCard, getContexts],
+  (cards, recentCard, contexts) => {
+    const { countryId: recentCountryId, commodityId: recentCommodityId } = recentCard.data[0];
+    console.log('c,', recentCountryId, recentCommodityId)
```
Comment (tomasmoose) [inline frontend/scripts/react-components/explore/explore.selectors.js:133]:
```suggestion
```

File: frontend/scripts/tests/explore/explore-selectors.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    const selector = getCardsWithRecentCard;
+    beforeEach(() => {
+      selector.resetAllComputations;
+      jest.resetAllMocks();
+    });
+
+    const setMock = (countryId, commodityId) => jest.fn(() => `${countryId}-${commodityId}`);
+    it('returns recent card if it has a valid context', () => {
+        Storage.prototype.getItem = setMock(validCountryId, validCommodityId);
```
Comment (tomasmoose) [inline frontend/scripts/tests/explore/explore-selectors.test.js:32]: Recommend adding this in the global jest config, so we don't have to care about local storage, 

```
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
};

global.localStorage = new LocalStorageMock;
``` 

This is basically all you need to implement the necessary requirements for the API

**Comment by Bluesmile82** [comment]
Hi! Thanks for the comments! I have added the global localStorage mocking. The second test is still not working. We might want to find out why but I have to merge as they are experiencing this bug and they need to test sandbox

**Comment by tomasmoose** [comment]
> Hi! Thanks for the comments! I have added the global localStorage mocking. The second test is still not working. We might want to find out why but I have to merge as they are experiencing this bug and they need to test sandbox

no worries, I would rather remove the test as now we probably wont fix it as its merged

**Pull request merged**